addons/hr_attendance/controllers/main.py

```diff
@@ -152,3 +159,29 @@ def systray_attendance(self, latitude=False, longitude=False):
     def user_attendance_data(self):
         employee = request.env.user.employee_id
         return self._get_employee_info_response(employee)
+
+    def has_password(self):
+        request.env.cr.execute(
+            f'SELECT COUNT(password) from res_users WHERE id={request.env.user.id} AND password IS NOT NULL',
+        )
+        return request.env.cr.fetchall()[0][0] != 0
+
+    @http.route('/hr_attendance/is_demo_state', type="json", auth="public")
+    def is_demo_state(self, token):
+        company = self._get_company(token)
+        if company:
+            employees = request.env['hr.employee'].sudo().search([])
+            return len(employees) == 1 and not employees[0].barcode
+
```

Comment:
```suggestion
    @http.route('/hr_attendance/is_demo_state', type="json", auth="public")
    def is_demo_state(self, token):
        company = self._get_company(token)
        if company:
            employees = request.env['hr.employee'].sudo().search([])
            return len(employees) == 1 and not employees[0].barcode
        return False
```

Reply:
Actually this doesn't reflect the fact you're on demo mode, what you you try to achieve here ? 

It looks like you check if there is only 1 employee (aka Admin) without any barcode.

If it's what you want, you should rename the route into something like ``is_fresh_db``.